### DISCUSSION TITLE
How to add GCC C++ Linker Flag AFTER cpp file ?

### DISCUSSION THREAD

**Discussion by ej-kon** [eclipse-cdt/cdt#542] [Q&A] [answered]
Hello , I am new in C++ and Eclipse CDT in this context and I have created a very simple CPP file that uses Imagick that compiles and works great from command line in Linux using :
g++ `Magick++-config --cxxflags --cppflags` -o "/path/to/executable/Test" /path/to/file/Test.cpp `Magick++-config --ldflags --libs`

I have added the `Magick++-config --cxxflags --cppflags`  part in (project) Properties > C/C++ Build / Settings / Tools Settings / GCC C++ Linker / Miscellaneous as Linker Flags and that part works great , what I haven't accomplished is to add the `Magick++-config --ldflags --libs` flag AFTER the cpp file.
Without it I am getting "undefined reference to" errors invoking GCC C++ Linker. 

I have tried many things:
I have tried to put it as library , it goes to the end but with -l in front of it making of course no sense and getting cannot find -l-L/usr/lib 
I have tried to put it next to `Magick++-config --cxxflags --cppflags` in the Linker Flag , but doesn't work , as I understand it needs to be declared after the cpp file. 
I have tried many attempts in C / C++ General / Paths and Symbols again with no lack. 
I can continue my work and compile the project through command line but I would really want to do it through Eclipse. 

Any help explaining what should I do to make it work would be great and highly appreciated.
Thank you in advance

**Comment by jld01** [comment, marked as answer]
I suggest you try separating the linker flags from the libraries as follows:

- GCC C++ Compiler > Miscellaneous > Other flags: `` `Magick++-config --cxxflags --cppflags` ``
- GCC C++ Linker > Miscellaneous > Linker flags: `` `Magick++-config --ldflags` ``
- GCC C++ Linker > Command line pattern: ``${COMMAND} ${FLAGS} ${OUTPUT_FLAG} ${OUTPUT_PREFIX}${OUTPUT} ${INPUTS} ${EXTRA_FLAGS} `Magick++-config --libs` ``

**Comment by ej-kon** [comment]
Thank you jld01 for your answer , that works great , and for helping me understand a bit more about it.
Now that I know it , it makes sense , but I wouldn't have figured it out myself. Again thank you.

**Comment by ej-kon** [comment]
How do I mark the discussion solved , but leave it to the threads so others that might have a similar problem view the answer ?

**Reply by jld01** [reply]
You can click the [Mark as answer](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/participating-in-a-discussion#marking-a-comment-as-an-answer) button to mark a discussion comment as being an answer to the original question.
